### PR TITLE
Fix more tests when run from a release branch

### PR DESCRIPTION
**Pull request by gibsondan**
Summary: These are only dev in master

Test plan: BK, push to release branch

## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: python_modules/dagit/dagit_tests/starlette/test_app.py [hunk 1]
```diff
@@ -20,9 +24,9 @@ def test_dagit_info(empty_app):
     response = client.get("/dagit_info")
     assert response.status_code == 200
     assert response.json() == {
-        "dagit_version": "dev",
-        "dagster_version": "dev",
-        "dagster_graphql_version": "dev",
+        "dagit_version": dagster_version,
```
Comment (prha) [inline python_modules/dagit/dagit_tests/starlette/test_app.py:27]: I think you want:
```
   "dagit_version": dagit_version,
   "dagster_version": dagster_version,
   "dagster_graphql_version": dagster_graphql_version,
```

**Review by prha — DISMISSED** [review]

**Comment by gibsondan** [comment]
good point. snuck in one more tiny fix that was failing in the RC

File: .buildkite/dagster-buildkite/dagster_buildkite/steps/dagster.py [hunk 2]
```diff
@@ -589,8 +589,9 @@ def graphql_python_client_backcompat_checks(version=SupportedPython.V3_8):
         StepBuilder("Backwards compat checks for the GraphQL Python Client")
         .on_integration_image(version)
         .run(
-            "pip install -e python_modules/dagster-graphql",
-            "pip install -e python_modules/automation",
+            "pip install -e python_modules/dagster[test] "
+            "-e python_modules/dagster-graphql "
+            "-e python_modules/automation",
```
Comment (prha) [inline .buildkite/dagster-buildkite/dagster_buildkite/steps/dagster.py:594]: either repeat `pip install` or make a single command?

Comment (gibsondan) [reply .buildkite/dagster-buildkite/dagster_buildkite/steps/dagster.py]: done

**Review by prha — APPROVED** [review]

**Pull request merged**